src/ws.nim

```diff
@@ -254,7 +250,7 @@ proc send*(ws: WebSocket, text: string, opcode = Opcode.Text): Future[void] {.as
       let data = frame[i ..< min(frame.len, i + maxSize)]
       discard await ws.tcpSocket.write(data)
```

Comment:
Write can still fail writing all the data, you should check if the bytes written match the data size, otherwise it's an error.

Reply:
Let's make the check and handle incomplete writes appropriately.